### ISSUE TITLE
Implement moderation features

### ISSUE THREAD

**Issue by georgeowell** [fraction/oasis#93]
Oasis should have moderation features implemented from the beginning rather than added on later as in Patchwork. Here are some ideas in no order:

- [ ] Content Warnings
- [ ] Follows
- [ ] Private follows
- [ ] Blocks
- [ ] Private blocks
- [ ] Block explanation UX (blocked because ....)
- [ ] Public block lists
- [ ] Hide content from feed (mute threads)
- [ ] Subscribe to other's public block lists

**Comment by jedahan** [comment]
This is an awesome, concise list of interesting features that would be cool to try out. I would also add to the list:

# Research
- [ ] Hire a UX researcher to help us understand and prioritize these features

I can recommend at least two that do good work, and at least get their rates and expected outcomes. LMK if I should spin this out into its own issue.

**Comment by christianbundy** [comment]
I'd vote to use the resources that @cinnamon-bun has already put together for safety and abuse prevention, if we can. I'm really impressed at how thoughtful they've been and it's miles above anything that I would've been able to put together.

**Comment by georgeowell** [comment]
Yep feel free to add to this @cinnamon-bun as you've already done amazing work in this area. Unfortunately I can't actually find the spreadsheet you made that lays out all of these ideas.

**Comment by georgeowell** [comment]
> This is an awesome, concise list of interesting features that would be cool to try out. I would also add to the list:
> # Research
> 
>     * [ ]  Hire a UX researcher to help us understand and prioritize these features
> 
> 
> I can recommend at least two that do good work, and at least get their rates and expected outcomes. LMK if I should spin this out into its own issue.

I am a UX tester btw. I would also just point out that I did already prioritise these features above already :heart:

**Comment by jedahan** [comment]
Ahh I was ignorant sorry if I implied otherwise. I saw the work and it reminded me of two people who do similar work ❤️

**Comment by cinnamon-bun** [comment]
@georgeowell Thanks for this list!  Oasis is the first open source project I've worked on where it's clear that accessibility and safety are deeply valued and it makes me really happy. ✨ Now that the groundwork of basic functionality has been filled out we can start turning more focus to these things.

I shared the spreadsheet with you on SSB just now.  There's also this document I wrote as an outline of all the safety needs.  It's a very long list!  Writing a social network is like Applied Utopia 101

https://hackmd.io/_x5i0jLPQ16TTaUaLt9Opg

@christianbundy We're approaching the end of the [Roadmap](https://github.com/fraction/oasis/blob/master/docs/roadmap.md)!  What are you interested in doing next?

**Comment by georgeowell** [comment]
I just had a another idea which I added: Hide content from feed using keywords

I know people use this quite a bit on Mastodon. It means you can totally bypass some new controversial topic that you don't want to read for whatever reason.

**Comment by Powersource** [comment]
> should have moderation features implemented from the beginning rather than added on later as an afterthought in Patchwork

As someone that's worked with blocks in patchwork I can really recommend doing this for at least blocks. So many corner cases when it' patched in on top of everything else...

**Comment by christianbundy** [comment]
I've opened an issue to ask about support for private blocks and follow here: https://github.com/ssbc/ssb-friends/issues/43

I don't think this is something we should try to implement ourselves until we at least reach out to others who care about this problem and see if we can share an implementation. I don't want to go the Patchwork path of all-custom-everything.

**Comment by jedahan** [comment]
How can I post with a content warning in oasis?

**Comment by christianbundy** [comment]
You can't, but they can be displayed.

**Comment by georgeowell** [comment]
I'm having a lot of content from people I have blocked showing up in my feeds. Why is that?

I also see replies to them from others which should be hidden.

**Comment by christianbundy** [comment]
@georgeowell That's a bug, could you make a new issue and document exactly where you're seeing content from blocked people?

**Comment by christianbundy** [comment]
@georgeowell I noticed you removed my 'enhancement' label and added 'question' instead. Is there a question I can answer to close this out?

**Comment by Powersource** [comment]
> Public block lists

Would be nice to hear from someone that knows more about this but I thiiink they tried this in mastodon and it ended poorly. So I think we should be careful with this/research it more before doing it.
